extra_geom/motors.py

```diff
@@ -0,0 +1,214 @@
+        tracker = cls(ref_geom)
+        if ref_motor_positions is None:
+            if hasattr(tracker.ref_geom, "motor_positions"):
+                ref_motor_positions = tracker.ref_geom.motor_positions
+            else:
+                raise ValueError(
+                    "There is no motor positions in reference geometry")
+
+        tracker.set_reference_positions(ref_motor_positions)
+        return tracker
+
+    def set_reference_positions(self, ref_motor_positions):
+        """Set reference motor positions.
+
+        Parameters
+        ----------
+        ref_motor_positions: array or list
+            New reference motor positions. The positions are expected
+            as array or list of the number of movable groups (quadrants)
+            by the number of motors per group.
+        """
+        ref_motor_positions = np.array(ref_motor_positions, copy=True)
+        if ref_motor_positions.shape != self.motor_position_shape:
+            raise ValueError(f"Expects array{self.motor_position_shape}: "
+                             f"{self.num_groups} groups moving by "
+                             f"{self.num_motors} motor each.")
+
+        self.ref_motor_positions = np.array(ref_motor_positions)
+
+    def set_motor_axes(self, new_motor_axes):
+        """Set the matrices of transformation motor positions in the positions
+        of detector panels.
+
+        ::
+            (h, v) - local motor coordinates
+            (x, y) - laboratory cooridnates (looking downstream)
+            (hx, hy) - the axis of horizontal motor in laboratory coordinates
+            (vx, vy) - the axis of vertical motor in laboratory coordinates
+
+            |x| _ | hx vx | |h|
+            |y| ‾ | hy vy | |v|
+
+        Parameters
+        ----------
+        new_motor_axes: array or list
+            New matrices of motor axes (transformation matrices). The matrices
+            are expected as three dimention array of the number of movable
+            groups (quadrants) by the number of panel coordinates (two) by
+            the number of motors per group.
+        """
+        new_motor_axes = np.array(new_motor_axes, copy=True)
+        if new_motor_axes.shape != self.motor_axes_shape:
+            raise ValueError(f"Expects array{self.motor_axes_shape}: "
+                             f"{self.num_groups} groups moving by "
+                             f"{self.num_motors} motor each.")
+        self.motor_axes = new_motor_axes
+
+    def geom_at(self, motor_positions):
+        """Update geometry according to the motor positions.
+
+        This updates the reference geometry according to the given
+        motor positions with respect the reference motor position.
+
+        Parameters
+        ----------
+        motor_positions: array or list
+            New motor positions as array of the number of movable groups
+            (quadrants) by the number of motor per group.
```

Comment:
The constant `1e-3` factor in the code implies that these positions are in mm. I'm partly tempted to say 'SI all the way' and put it in metres, but it looks like mm is how motor positions are usually recorded.

The units should definitely be mentioned in the docstring, though - both here and below in `move_geom_by`.